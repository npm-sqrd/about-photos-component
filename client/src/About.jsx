import React from 'react';
import PropTypes from 'prop-types';

const About = (props) => {
  return (
    <div>
      <div className="restaurant">
        About {props.restaurant}
      </div>
      <div className="description">
        {props.description}
      </div>
    </div>
  );
};

About.propTypes = {
  restaurant: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default About;
