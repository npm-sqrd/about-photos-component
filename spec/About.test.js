import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import sample from '../data/sampleData.js';

import About from '../client/src/About.jsx';

Enzyme.configure({ adapter: new Adapter() });

describe('About Component', () => {
  test('should render correctly', () => {
    expect(shallow(<About photo={sample[0].photo} />)).toMatchSnapshot();
  });
});

describe("rendered `About`", () => {
  it("received three props", () => {
    const component = shallow(<About photo={sample[0].photo} />);
    expect(component.props().children.length).toBe(3);
  });
});
