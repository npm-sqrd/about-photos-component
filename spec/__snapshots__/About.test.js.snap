// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <About />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="restaurant"
        >
          About
        </div>,
        <div
          className="description"
        >
          description
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "About",
          "className": "restaurant",
        },
        "ref": null,
        "rendered": "About",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "description",
          "className": "description",
        },
        "ref": null,
        "rendered": "description",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="restaurant"
          >
            About
          </div>,
          <div
            className="description"
          >
            description
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "About",
            "className": "restaurant",
          },
          "ref": null,
          "rendered": "About",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "description",
            "className": "description",
          },
          "ref": null,
          "rendered": "description",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
